docs/.vitepress/theme/ModalDemo.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <Button :action="showModal">Show Modal</Button>
+  <ModalReport
+    ref="reportModal"
+    :reportTypes="['cringitude', 'rudeness', 'notgamer', 'windowsuser']"
+  >
+  </ModalReport>
+</template>
+
+<script setup>
+import { ModalReport, Button } from '@/components'
+</script>
+
+<script>
+import { defineComponent } from 'vue'
+
+export default defineComponent({
+  methods: {
+    showModal() {
+      this.$refs.reportModal.modal.show()
+    },
+  },
+})
+</script>
+
+<style scoped lang="scss"></style>
```

Comment:
I don't think this needs it's own component. It can just be inline in the .md file. It should work (we do this elsewhere)